### ISSUE TITLE
sia 1.2.1 - Could not resize folder: unable to migrate all sectors

### ISSUE THREAD

**Issue by DcyMatrix** [NebulousLabs/Sia#1829]
It all started with me having a RAID1 BTRFS raid setup, I thought (wrongly) that I would have at least 3.6TB for the Sia Hosting folder.

But when I upgraded to Sia 1.2.1, and Sia converted the old 'many files' to one huge file and a small one. The BTRFS showed its stubbornness and stopped working correctly with a Sia data file of about 3.2TB
Absolutely no more space on the BRTFS raid1, I could not convert the raid1 to a single either. No more space I was told by BTRFS.

So I borrowed some external hdd's setup a single (JBOD) BTRFS and got the 3.2TB file moved from the original raid1.

I then converted my original raid1 to single (JBOD) plus added bigger and more hdds to the array.
I then copy'ed back the 2 Sia host files. So far so good.

at this point, siad seems to be working fine, receiving contracts, and data is being added now to the siad host.

And now the trouble begins, when I try to resize the sia share folder with the command:
`./siac host folder resize /media/sia-hosting 2500GB `

I get this reply after a while:
`Could not resize folder: unable to migrate all sectors`

But according to:
`./siac host -v`

There is only some 485GB in use of the 3.6TB share. And the disk space available should be plenty to make a resized copy of the 2 sia host files at a size of 2500GB

So to recap, skewed up, thought I had 3.6TB for a Sia host folder, doing conversion that came with Sia 1.2.1 my BTRFS filesystem filled up with a 'incomplete' (I assume) 3.2TB file, it should have been 3.6TB as far as I know. But not enough space to make it that big.
Can now not resize the 3.2TB file to a 2.5TB file due to the error:
`Could not resize folder: unable to migrate all sectors`

If you need logs, let me know what you need if you want to look into this.

**Comment by tlightsky** [comment]
same problem:
![image](https://cloud.githubusercontent.com/assets/429211/26385846/fe374870-4074-11e7-8ed1-97280b1af8d3.png)

from 2TB to 1TB (63GB used)

**Comment by DavidVorick** [comment]
You need empty storage folders when downsizing a storage folder.

**Comment by DcyMatrix** [comment]
@DavidVorick ah okay :)

Can you help me with what the command would look like then?
Right now I have:
`./siac host folder resize /media/sia-hosting 2500GB`

Thank you in advance!

**Comment by DavidVorick** [comment]
You need to add an additional folder with probably a hundred gb or so to collect spillover. Then you can remove it.

Sia doesn't currently support shrinking a single storage folder. We will add it soon, but the code to do so was non-trivial

**Comment by DcyMatrix** [comment]
@DavidVorick Thank you! I will close this now :-D

**Comment by chevdor** [comment]
Just to add a little doc for those who may need to do the same.

I had a 5GB allowance in a STORAGE folder. Less than 1GB was used. 
I wanted to reduce to 4GB and successfully did the following:
```
mkdir STORAGE01
mkdir STORAGE02
./siac host folder add ./STORAGE01 2GB      // ensure the size you pick here is more than the amount of data you currently store for your users
./siac host folder remove ./STORAGE
./siac host folder add ./STORAGE02 2GB

```
